### ISSUE TITLE
Remove circular dependency

### ISSUE THREAD

**Issue by doodlewind** [toeverything/blocksuite#2689]
Related: #2686 - we can handle this in a progressive way, removing circular imports using multiple PRs.

**Comment by himself65** [comment]
```sh
✗ madge --circular ./packages/blocks/src/index.ts
Processed 297 files (944ms) (23 warnings)

✖ Found 46 circular dependencies!

1) __internal__/utils/common-operations.ts > __internal__/utils/query.ts > __internal__/utils/selection.ts
2) __internal__/utils/query.ts > __internal__/utils/selection.ts
3) __internal__/service.ts > __internal__/service/index.ts
4) __internal__/service.ts > __internal__/service/index.ts > __internal__/service/json2block.ts
5) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts
6) __internal__/clipboard/edgeless-clipboard.ts > __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts
7) __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts
8) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts
9) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts
10) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts
11) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts
12) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts
13) __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/format-quick-bar/index.ts > components/format-quick-bar/format-bar-node.ts > page-block/utils/const.ts
14) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/format-quick-bar/index.ts > components/format-quick-bar/format-bar-node.ts > page-block/utils/index.ts > page-block/utils/bind-hotkey.ts
15) page-block/utils/index.ts > page-block/utils/bind-hotkey.ts
16) page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/format-quick-bar/index.ts
17) __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/index.ts > components/import-page/index.ts > components/import-page/import-page.ts
18) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/index.ts > components/slash-menu/index.ts > components/slash-menu/config.ts
19) page-block/default/utils.ts > components/index.ts > components/slash-menu/index.ts > components/slash-menu/config.ts
20) page-block/index.ts > page-block/default/default-page-block.ts
21) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts
22) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts
23) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts
24) bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts
25) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > code-block/index.ts > code-block/code-block.ts > __internal__/rich-text/rich-text.ts > __internal__/rich-text/keyboard.ts
26) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > code-block/index.ts > code-block/code-block.ts
27) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts
28) database-block/common/view-manager.ts > database-block/database-model.ts
29) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts > database-block/database-service.ts
30) database-block/table/table-view.ts > database-block/table/components/column-header/column-header.ts > database-block/table/components/column-header/database-header-column.ts
31) database-block/table/table-view.ts > database-block/table/components/column-header/column-width-drag-bar.ts
32) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts > database-block/table/table-view.ts > database-block/table/components/selection/selection.ts
33) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts > database-block/table/table-view.ts > database-block/table/components/toolbar/toolbar.ts
34) database-block/common/literal/index.ts > database-block/common/literal/array-literal.ts
35) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > divider-block/index.ts > divider-block/divider-block.ts
36) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > image-block/index.ts > image-block/image-block.ts
37) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > list-block/index.ts > list-block/list-block.ts
38) index.ts > list-block/index.ts > list-block/list-block.ts
39) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > note-block/index.ts > note-block/note-block.ts
40) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > note-block/index.ts > note-block/note-block.ts > note-block/note-service.ts
41) page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts
42) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > paragraph-block/index.ts > paragraph-block/paragraph-block.ts
43) bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > preset/index.ts
44) page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > preset/index.ts
45) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/page-service.ts
46) __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts

➜  blocksuite git:(himself65/fix-circular) ✗ git stash
Saved working directory and index state WIP on himself65/fix-circular: 6c5ed323 fix: code
➜  blocksuite git:(himself65/fix-circular) madge --circular ./packages/blocks/src/index.ts
Processed 297 files (1s) (23 warnings)

✖ Found 45 circular dependencies!

1) __internal__/utils/common-operations.ts > __internal__/utils/query.ts > __internal__/utils/selection.ts
2) __internal__/utils/query.ts > __internal__/utils/selection.ts
3) __internal__/service.ts > __internal__/service/index.ts
4) __internal__/service.ts > __internal__/service/index.ts > __internal__/service/json2block.ts
5) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts
6) __internal__/clipboard/edgeless-clipboard.ts > __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts
7) __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts
8) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts
9) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts
10) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts
11) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts
12) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts
13) __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/format-quick-bar/index.ts > components/format-quick-bar/format-bar-node.ts > page-block/utils/const.ts
14) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/format-quick-bar/index.ts > components/format-quick-bar/format-bar-node.ts > page-block/utils/index.ts > page-block/utils/bind-hotkey.ts
15) page-block/utils/index.ts > page-block/utils/bind-hotkey.ts
16) page-block/index.ts > page-block/default/default-page-block.ts > page-block/default/utils.ts > components/format-quick-bar/index.ts
17) page-block/index.ts > page-block/default/default-page-block.ts
18) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts
19) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts
20) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts
21) bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts
22) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > code-block/index.ts > code-block/code-block.ts > __internal__/rich-text/rich-text.ts > __internal__/rich-text/keyboard.ts
23) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > code-block/index.ts > code-block/code-block.ts
24) __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > components/index.ts > components/import-page/index.ts > components/import-page/import-page.ts
25) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > components/index.ts > components/slash-menu/index.ts > components/slash-menu/config.ts
26) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts
27) database-block/common/view-manager.ts > database-block/database-model.ts
28) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts > database-block/database-service.ts
29) database-block/table/table-view.ts > database-block/table/components/column-header/column-header.ts > database-block/table/components/column-header/database-header-column.ts
30) database-block/table/table-view.ts > database-block/table/components/column-header/column-width-drag-bar.ts
31) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts > database-block/table/table-view.ts > database-block/table/components/selection/selection.ts
32) __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > database-block/index.ts > database-block/database-block.ts > database-block/table/table-view.ts > database-block/table/components/toolbar/toolbar.ts
33) database-block/common/literal/index.ts > database-block/common/literal/array-literal.ts
34) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > divider-block/index.ts > divider-block/divider-block.ts
35) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > image-block/index.ts > image-block/image-block.ts
36) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > list-block/index.ts > list-block/list-block.ts
37) index.ts > list-block/index.ts > list-block/list-block.ts
38) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > note-block/index.ts > note-block/note-block.ts
39) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > note-block/index.ts > note-block/note-block.ts > note-block/note-service.ts
40) page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts
41) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > paragraph-block/index.ts > paragraph-block/paragraph-block.ts
42) bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > preset/index.ts
43) page-block/index.ts > page-block/edgeless/edgeless-page-block.ts > index.ts > preset/index.ts
44) __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts > __internal__/clipboard/page-clipboard.ts > page-block/index.ts > page-block/page-service.ts
45) __internal__/clipboard/utils/commons.ts > __internal__/content-parser/index.ts > __internal__/service.ts > services.ts > bookmark-block/index.ts > bookmark-block/bookmark-block.ts > bookmark-block/components/bookmark-toolbar.ts > bookmark-block/components/bookmark-operation-popper.ts > __internal__/clipboard/index.ts
```

**Comment by doodlewind** [comment]
@himself65 is this fixed after #3451 landed?

**Comment by himself65** [comment]
I only fixed 2 references

**Comment by himself65** [comment]
There are still 45

**Comment by Saul-Mirone** [comment]
19 now

**Comment by doodlewind** [comment]
Closed by #4214
